dvc/repo/__init__.py

```diff
@@ -174,7 +174,8 @@ def _ignore(self):
 
     def check_modified_graph(self, new_stages):
         """Generate graph including the new stage to check for errors"""
-        self._collect_graph(self.stages + new_stages)
+        if not getattr(self, "_skip_graph_checks", False):
+            self._collect_graph(self.stages + new_stages)
```

Comment:
Maybe set it in to `False` in `__init__` along with a nice comment explaining what it is for? Also, we need a test here.

Reply:
Prefer not touching constructor, keep it in single place. Added an explanation comment and a test.